operator/beiboot/provider/k3s/__init__.py

```diff
@@ -0,0 +1,246 @@
+import re
+from datetime import timedelta
+from typing import List
+
+import kopf
+import kubernetes as k8s
+
+from beiboot.configuration import BeibootConfiguration, ClusterConfiguration
+from beiboot.provider.abstract import AbstractClusterProvider
+from beiboot.utils import exec_command_pod
+
+from .utils import (
+    create_k3s_server_workload,
+    create_k3s_agent_workload,
+    create_k3s_kubeapi_service,
+)
+
+core_api = k8s.client.CoreV1Api()
+
+
+class K3s(AbstractClusterProvider):
```

Comment:
Seems like we are duplicating code from getdeck here. Is this intended? Could make this hard to maintain.